test/src/lib.rs

```diff
@@ -418,6 +424,44 @@ mod tests {
         }
     }
 
+    #[test]
+    #[ignore]
+    fn cwe_252() {
+        let mut error_log = Vec::new();
+        let mut tests = linux_test_cases("cwe_252", "CWE252");
+        let num_expected_occurences = 9;
+        let num_expected_occurences_lkm = 1;
+
+        tests.extend(lkm_test_cases("cwe_252", "CWE252"));
+
+        mark_architecture_skipped(&mut tests, "ppc64");
+        mark_architecture_skipped(&mut tests, "ppc64le");
+        mark_skipped(&mut tests, "aarch64", "gcc");
```

Comment:
Do you know the reason why it fails for aarch64-gcc? Or maybe I should ask why it fails for gcc instead... I am mainly interested in whether turning on optimizations has something to do with it or not.

Reply:
For the spurious function entries, `-O0` might again help. I know, you prefer testing on optimized code since it is more similar to real world code than `-O0` code. I think in the future, it would be better to have two sets of acceptance tests: One set of very simple test (like the current ones) that we test for each ISA-compiler pairing. Probably using `-O0` just to reduce the number of strange optimization edge cases we have to handle just for the CI. And one set of more complex tests that we only test for one or two ISA-compiler pairings (with `-O2`) where we can use knowledge about how the compiler will optimize the code to make the tests stable enough to run in the CI on each pull request.

For the difference in the extra layer of indirection in gcc we already talked a bit about it. Maybe there is no good solution, maybe we can change the check so that it reports the CWE always at the the end of the callee? I am content with blacklisting the test case for now (like you did). We could revisit the problem at a later time when we have some more experience on real-world binaries.